### PR TITLE
changed constructors of event classes to public 

### PR DESCRIPTION
**Pull request by jzarnikov**
so they can be instatiated in tests of users of this library.

https://github.com/messenger4j/messenger4j/issues/2

### CODE REVIEW

File: src/main/java/com/github/messenger4j/send/templates/ListTemplate.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+        return "ListTemplate{" +
+                "topElementStyle=" + topElementStyle +
+                ", buttons=" + buttons +
+                ", elements=" + elements +
+                '}';
+    }
+
+    public static final class Builder {
+
+        public static final int BUTTONS_LIMIT = 1;
```
Comment (maxgrabenhorst) [inline src/main/java/com/github/messenger4j/send/templates/ListTemplate.java]: please make private

File: src/main/java/com/github/messenger4j/send/templates/ListTemplate.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        private final TopElementStyle topElementStyle;
+
+        private List<Button> buttons;
+        private List<Element> elements;
+
+        public Builder(TopElementStyle topElementStyle) {
+            this.topElementStyle = topElementStyle;
```
Comment (maxgrabenhorst) [inline src/main/java/com/github/messenger4j/send/templates/ListTemplate.java]: According the Messenger spec it is not required to provide a top element style (defaults to compact) but I think its a good practice to decide first what type of list template you want to create. Just a remark. No changes needed here.

File: src/main/java/com/github/messenger4j/send/templates/ListTemplate.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        }
+
+        private Builder elements(List<Element> elements) {
+            this.elements = elements;
+            return this;
+        }
+
+        public ListTemplate build() {
+            Assert.notNullOrEmpty(this.elements, "elements");
+            Assert.isTrue(topElementStyle == TopElementStyle.COMPACT || elements.get(0).imageUrl != null,
```
Comment (maxgrabenhorst) [inline src/main/java/com/github/messenger4j/send/templates/ListTemplate.java]: would prefer to check existence of imageUrl with Assert.notNullOrBlank

File: src/main/java/com/github/messenger4j/send/templates/ListTemplate.java [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            @SerializedName("webview_height_ratio")
+            private final WebviewHeightRatio webviewHeightRatio;
+
+            public DefaultAction(Builder builder) {
+                this.url = builder.url;
+                this.webviewHeightRatio = builder.webviewHeightRatio;
+                this.type = Button.ButtonType.URL;
+            }
+
+            public enum WebviewHeightRatio {
```
Comment (maxgrabenhorst) [inline src/main/java/com/github/messenger4j/send/templates/ListTemplate.java]: WebviewHeightRatio already exists in UrlButton. Please reuse. Could be reasonable to move the inner class to upper level.

File: src/main/java/com/github/messenger4j/send/templates/ListTemplate.java (hunk 1 above)
Comment (jzarnikov) [inline src/main/java/com/github/messenger4j/send/templates/ListTemplate.java]: done

File: src/main/java/com/github/messenger4j/send/templates/ListTemplate.java (hunk 4 above)
Comment (jzarnikov) [inline src/main/java/com/github/messenger4j/send/templates/ListTemplate.java]: I was considering it but then I decided to use a separete enum. I've changed it now. 
The spec explicitly says that the DefaultAction has the same attributes as UrlButton so I guess it is ok if we reference it in our code.

**Comment by jzarnikov** [comment]
All the issues should be fixed now.

**Pull request merged**